test/e2e/framework/framework.go

```diff
@@ -0,0 +1,249 @@
+func NewDefaultFramework(baseName string) *Framework {
+    options := Options{
+        ClientQPS:   20,
+        ClientBurst: 50,
+    }
+    return NewFramework(baseName, options, nil)
+}
+
+// NewFramework creates a test framework.
+func NewFramework(baseName string, options Options, client *kubeclientset.Clientset) *Framework {
+    f := &Framework{
+        BaseName:                 baseName,
+        Options:                  options,
+        ClientSet:                client,
+    }
+
+    ginkgo.BeforeEach(f.BeforeEach)
+    ginkgo.AfterEach(f.AfterEach)
+
+    return f
+}
+
+
+func (f *Framework) BeforeEach() {
+    // The fact that we need this feels like a bug in ginkgo.
+    // https://github.com/onsi/ginkgo/issues/222
+    f.cleanupHandle = AddCleanupAction(f.AfterEach)
+
+    if f.ClientSet == nil {
+        ginkgo.By("Creating a kubernetes client")
+
+        restConfig, _, err := loadConfig(TestContext.KubeConfig, TestContext.KubeContext)
+        ExpectNoError(err)
+        testDesc := ginkgo.CurrentGinkgoTestDescription()
+        if len(testDesc.ComponentTexts) > 0 {
+            componentTexts := strings.Join(testDesc.ComponentTexts, " ")
+            restConfig.UserAgent = fmt.Sprintf(
+                "%v -- %v",
+                rest.DefaultKubernetesUserAgent(),
+                componentTexts)
+        }
+
+        Expect(err).NotTo(HaveOccurred())
+
+        restConfig.QPS = f.Options.ClientQPS
+        restConfig.Burst = f.Options.ClientBurst
+
+        if f.Options.GroupVersion != nil {
+            restConfig.GroupVersion = f.Options.GroupVersion
+        }
+
+        f.ClientSet, err = kubeclientset.NewForConfig(restConfig)
+        Expect(err).NotTo(HaveOccurred())
+
+
+        // node.k8s.io is based on CRD, which is served only as JSON
+        jsonConfig := restConfig
+        jsonConfig.ContentType = "application/json"
+        Expect(err).NotTo(HaveOccurred())
+
+        // create scales getter, set GroupVersion and NegotiatedSerializer to default values
+        // as they are required when creating a REST client.
+        if restConfig.GroupVersion == nil {
+            restConfig.GroupVersion = &schema.GroupVersion{}
+        }
+        if restConfig.NegotiatedSerializer == nil {
+            restConfig.NegotiatedSerializer = scheme.Codecs
+        }
+
+    }
+
+    if !f.SkipNamespaceCreation {
+        ginkgo.By(fmt.Sprintf("Building a namespace api object, basename %s", f.BaseName))
+        namespace  := f.CreateNamespace(f.BaseName, map[string]string{
+            "e2e-framework": f.BaseName,
+        })
+
+        f.Namespace = namespace
+        f.UniqueName = f.Namespace.GetName()
+    } else {
+        // not guaranteed to be unique, but very likely
+        f.UniqueName = fmt.Sprintf("%s-%08x", f.BaseName, rand.Int31())
+    }
+
+}
+
+
+func deleteNamespace(client kubeclientset.Interface, namespaceName string) error {
+    orphanDependents := false
```

Comment:
`orphanDependents` is deprecated, and passing nil for the options to `Delete` should be fine.

Reply:
<img width="634" alt="Screenshot 2019-05-30 at 11 36 40" src="https://user-images.githubusercontent.com/1176434/58623920-4a6cdb80-82cf-11e9-9de2-4228beb3187f.png">

Oh, I must pay attention, the IDE was telling me.